lib/hammer_cli_katello/content_view_purge.rb

```diff
@@ -0,0 +1,54 @@
+module HammerCLIKatello
+  class ContentViewPurgeCommand < HammerCLIKatello::Command
+    command_name "purge"
+    desc "Delete old versions of a content view"
+
+    option "--id", "ID", _("content view numeric identifier"), required: true
+    option "--count", "COUNT", _("count of unused versions to keep"),
+           default: 3, format: HammerCLI::Options::Normalizers::Number.new
+
+    def resource_content_views
+      HammerCLIForeman.foreman_resource(:content_views)
+    end
+
+    def resource_content_view_versions
+      HammerCLIForeman.foreman_resource(:content_view_versions)
+    end
+
+    def execute
+      if option_count < 0
+        output.print_error _("Invalid value for --count option: value must be 0 or greater.")
+        HammerCLI::EX_USAGE
+      else
+        view = resource_content_views.call(:show, 'id' => option_id)
+
+        # Sort versions by the number (1.8, 1.10, 2.4, ...)
+        all_versions = view['versions'].sort_by do |version|
+          [version[:major], version[:minor]]
+        end
+
+        # Keep only versions which are not used
+        old_versions = all_versions.select do |v|
+          v["environment_ids"].empty?
+        end
+
+        # Check if there is something to do
+        if option_count >= old_versions.size
+          output.print_error _("No versions to delete.")
+          HammerCLI::EX_NOT_FOUND
+        else
+          versions_to_purge = old_versions.slice(0, old_versions.size - option_count)
+
+          versions_to_purge.each do |v|
+            HammerCLIForeman.record_to_common_format(
+              resource_content_view_versions.call(:destroy, 'id' => v["id"])
+            )
+            print_message _("Version '%{version}' of content view '%{view}' deleted.") %
```

Comment:
Since the destroy call is an async operation, I wonder if we should indicate that it has been scheduled vs deleted?

In the future, it may be useful to enhance it so that it monitors the destroy through completion.  (e.g. using foreman tasks)